Lib/test/test_subclassinit.py

```diff
@@ -228,11 +225,19 @@ class MyClass(metaclass=MyMeta, otherarg=1):
                             dict(metaclass=MyMeta, otherarg=1))
 
         class MyMeta(type):
-            def __init__(self, name, bases, namespace, otherarg):
+            def __init__(self, name, bases, namespace, otherarg=0):
                 super().__init__(name, bases, namespace)
+                self.otherarg = otherarg
 
-        with self.assertRaises(TypeError):
-            class MyClass(metaclass=MyMeta, otherarg=1):
+        class MyClass(metaclass=MyMeta):
+            def __init_subclass__(cls, otherarg):
+                pass
+
+        with self.assertRaisesRegex(
+                TypeError,
+                ".*MyClass\.__init_subclass__.. missing 1 required positional argument: .otherarg.",
+            ):
+            class MyOtherClass(MyClass, otherarg=2):
                 pass
 
```

Comment:
Now we are testing that `__init__` can cause problems for `__init_subclass__` if it doesn't pass through the required keyword arguments.